### DISCUSSION TITLE
Use with Digitech FS3X footswitch

### DISCUSSION THREAD

**Discussion by jabarlee** [bxparks/AceButton#74] [Q&A]
![fs3x-wiring](https://user-images.githubusercontent.com/82775953/115156466-14849c00-a08d-11eb-906e-5313aa2aaecc.jpg)
Hi and thanx for this wonderful library.
I'm new to coding and Arduino and though I've managed to do quite some things for my midi project, this one I can't seem to solve.
I want to use Digitech's FS3X footswitch in order to send midi sysex messages to my device. I have attached the wiring and, as you can see, it uses a combination of tip/ring to send the signal of the 3rd button:
Button1: tip, Button2: ring, Button3: tip & ring

Whatever I may have tried, it always recognizes button3 press as 2 separate presses from button1 and button2. Is there any workaround to this ?

Thanx in advance!

**Comment by bxparks** [comment]
Hi, not enough information. What processor are you using? How is this circuit connected to the pins? Are you using pull-down or pull-up? How are you configuring the `ButtonConfig`?

**Reply by jabarlee** [reply]
I'm sorry, you're right.
I'm using an old Duemilanove/Diecimila Arduino board (ATmega168)
I connect the footswitch to the board via TRS jack: tip connects directly to pin10, ring to pin 11, sleeve(GND) to gnd
I'm using pull-up and actually what I did is edit your "TwoButtonsUsingOneButtonConfig" example as a test, I didn't mess with ButtonConfig at all, all the editing was at the event handler.

**Reply by bxparks** [reply]
Got it, I kinda had a feeling that was roughly what you were doing, but we save time by making sure.

It looks like the footswitch is using binary encoding, which allows it to support 3 switches using only 2 data lines. I added support for that relatively recently (...hmm, looks like it was over a year ago). Take a look at https://github.com/bxparks/AceButton#BinaryEncodedButtons, and the additional document that I linke from there, https://github.com/bxparks/AceButton/blob/develop/docs/binary_encoding/README.md. You can use the `Encoded4To2ButtonConfig` class in your case.  I created an example here: https://github.com/bxparks/AceButton/blob/develop/examples/Encoded8To3Buttons/Encoded8To3Buttons.ino

Hope that helps.

**Reply by jabarlee** [reply]
Oh, the versatility!
This is absolutely fantastic. I didn't read the whole documentation when I started, believing that what I was looking for was pretty straightforward. It turns out that it isn't but you had already covered every possible aspect!

Thank you again for the quick response and ofc for that brilliant piece of code, I got my setup working in a matter of 30'
